File: backend/src/main/java/com/tasklion/backend/service/TasklionUserService.java
package com.tasklion.backend.service;

import com.tasklion.backend.model.FieldValueModel;
import com.tasklion.backend.model.TasklionUserModel;

public interface TasklionUserService {

    boolean isExists(FieldValueModel requestModel);

    TasklionUserModel getTasklionUser(String username);

}

File: frontend/src/app/services/tasklion-user/tasklion-user.service.ts
import {Injectable} from '@angular/core';
import {Observable} from "rxjs";
import {ApiService} from "../../shared/services/api/api.service";
import {ApiUrlConstant} from "../../shared/constants/api-url.constant";
import {FieldValueModel} from "../../shared/models/field-value.model";
import {ApiResponseModel} from "../../shared/models/api/api-response.model";

@Injectable({
  providedIn: 'root'
})
export class TasklionUserService extends ApiService {

  public isExists(fieldValueModel: FieldValueModel): Observable<ApiResponseModel<any>> {
    return this.post<ApiResponseModel<any>>(ApiUrlConstant.IS_TASKLION_USER_EXISTS, fieldValueModel);
  }

}
